src/test/java/teammates/sqlui/webapi/JoinCourseActionTest.java

```diff
@@ -0,0 +1,266 @@
+
+    @Test
+    void testExecute_studentDoesNotExist_throwsEntityNotFoundException() throws EntityAlreadyExistsException,
+            InvalidParametersException, EntityDoesNotExistException {
+        loginAsUnregistered("unreg-student");
+
+        when(mockLogic.getStudentByRegistrationKey("invalid-reg-key")).thenReturn(stubStudent);
+        when(mockLogic.joinCourseForStudent("invalid-reg-key", "unreg-student"))
+                .thenThrow(EntityDoesNotExistException.class);
+        String[] params = {
+                Const.ParamsNames.REGKEY, "invalid-reg-key",
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT,
+        };
+        verifyEntityNotFound(params);
+    }
+
+    @Test
+    void testExecute_studentInvalidAccount_errorMessage() throws EntityAlreadyExistsException,
+            InvalidParametersException, EntityDoesNotExistException {
+        loginAsUnregistered("unreg-student");
+        when(mockLogic.getStudentByRegistrationKey("invalid-reg-key")).thenReturn(stubStudent);
+        when(mockLogic.joinCourseForStudent("invalid-reg-key", "unreg-student"))
+                .thenThrow(InvalidParametersException.class);
+        String[] params = {
+                Const.ParamsNames.REGKEY, "invalid-reg-key",
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT,
+        };
+        JoinCourseAction action = getAction(params);
+        JsonResult jsonResult = getJsonResult(action, 500);
+        assertEquals(500, jsonResult.getStatusCode());
+    }
+
+    @Test
+    void testExecute_validInstructorRegKey_success() throws EntityAlreadyExistsException,
+            InvalidParametersException, EntityDoesNotExistException {
+        loginAsUnregistered("unreg-instructor");
+
+        when(mockLogic.getInstructorByRegistrationKey("registered-key-instructor")).thenReturn(stubInstructor);
+        when(mockLogic.joinCourseForInstructor("registered-key-instructor", "unreg-instructor"))
+                .thenReturn(stubInstructor);
+        when(mockLogic.getCourse(stubInstructor.getCourseId())).thenReturn(stubCourse);
+        when(mockSqlEmailGenerator.generateUserCourseRegisteredEmail(stubInstructor.getName(), stubInstructor.getEmail(),
+                "unreg-instructor", true, stubCourse)).thenReturn(stubEmailWrapper);
+        String[] params = {
+                Const.ParamsNames.REGKEY, "registered-key-instructor",
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR,
+        };
+        JoinCourseAction action = getAction(params);
+        JsonResult jsonResult = getJsonResult(action);
+        MessageOutput messageOutput = (MessageOutput) jsonResult.getOutput();
+        assertEquals("Instructor successfully joined course", messageOutput.getMessage());
+        verify(mockEmailSender, times(1)).sendEmail(stubEmailWrapper);
+    }
+
+    @Test
+    void testExecute_instructorAlreadyExists_throwsEntityAlreadyExistsException() throws EntityAlreadyExistsException,
```

Comment:
Same as above. should this test case be named `..._throwsInvalidParametersException()` instead?